lib/Serialization/Deserialization.cpp

```diff
@@ -5441,7 +5440,7 @@ llvm::Error DeclDeserializer::deserializeDeclCommon() {
           peersAndMembers.push_back(MacroIntroducedDeclName(*kind, identifier));
         }
         Attr = DeclarationAttr::create(
-            ctx, SourceLoc(), SourceRange(), macroContext,
+            ctx, SourceLoc(), SourceRange(), role,
             peersAndMembersRef.take_front(numPeers),
             peersAndMembersRef.take_back(numMembers),
             isImplicit);
```

Comment:
Missing deserialization for `decls_block::Declaration_DECL_ATTR`?

Reply:
Oh, good catch! I better write a test. Also AST printing is missing